vscode_extension/src/aacExecutableWrapper.ts

```diff
@@ -0,0 +1,155 @@
+        if(commandName) {
+            getAacCommandArgs(commandName).then(async (commandArguments) => {
+                await getCommandArgUserInput(commandArguments);
+                execAacShellCommand(commandName, commandArguments).then(output => {
+                    outputChannel.appendLine(output);
+                    outputChannel.show();
+                });
+            });
+        }
+    });
+}
+
+/**
+ * Gets the version of the currently installed AaC package
+ * @returns string of the version number or null if not installed.
+ */
+export async function getAaCVersion(): Promise<string|null> {
+
+    let aacVersionOutput: string = "";
+    try {
+        aacVersionOutput = await execAacShellCommand("version");
+    } catch (exception) {
+        // TODO: Change to logging or alternative
+    }
+
+    const regExp = /([0-9]+\.*){3}/;
+    const versionMatch = regExp.exec(aacVersionOutput);
+    return versionMatch ? versionMatch[0] : null;
+}
+
+async function getCommandArgUserInput(commandArguments: CommandArgument[]) {
+
+    let unansweredCommandArguments: CommandArgument[] = commandArguments.filter(argument => { return (argument.userResponse === undefined); } );
+
+    if (unansweredCommandArguments.length > 0) {
+        let argumentToPromptFor = unansweredCommandArguments[0];
+        const dialogBoxOptions: OpenDialogOptions = {
+            title: argumentToPromptFor.name,
+            canSelectMany: false
+        };
+        const inputBoxOptions: InputBoxOptions = {
+            title: argumentToPromptFor.name,
+            prompt: argumentToPromptFor.description,
+        };
+
+        if (argumentToPromptFor.description.toLowerCase().includes("path")){
+            let fileUri: Uri[] | undefined = await window.showOpenDialog(dialogBoxOptions);
+            argumentToPromptFor.userResponse = fileUri ? fileUri[0]?.path : "";
+
+        } else {
+            argumentToPromptFor.userResponse = await window.showInputBox(inputBoxOptions);
+        }
+    } else if(unansweredCommandArguments.length > 1) {
+        getCommandArgUserInput(commandArguments);
+    }
+}
+
+async function getAacCommandNames(): Promise<string[]> {
+    const aacHelpOutput = await execAacShellCommand("-h");
+    return parseTaskNamesFromHelpCommand(aacHelpOutput);
+}
+
+/**
+ * @returns list of Command arguments
+ */
+async function getAacCommandArgs(aacCommandName: string): Promise<CommandArgument[]> {
+    const aacHelpOutput = await execAacShellCommand(`${aacCommandName} -h`);
+    return parseTaskArgsFromHelpCommand(aacHelpOutput);
+}
+
+/**
+ * Executes AaC commands with arguments.
+ * @param command - AaC command name
+ * @param commandArgs - command arguments and
+ * @returns
+ */
+async function execAacShellCommand(command: string, commandArgs: CommandArgument[] = []): Promise<string> {
+
+    let commandArgsArray = ["aac", command, ...(commandArgs.map(argument => argument.userResponse))];
```

Comment:
I think we should use the user-defined `aacPath` configuration item and, only if it's not provided, default to a globally available `aac` tool.

```suggestion
    const aacPath = getConfigurationItem("aacPath") ?? "aac";
    const commandArgsArray = [aacPath, command, ...(commandArgs.map(argument => argument.userResponse))];
```